include/ada/scheme-inl.h

```diff
@@ -16,46 +16,50 @@ namespace ada::scheme {
 namespace details {
 // for use with is_special and get_special_port
 // Spaces, if present, are removed from URL.
-constexpr std::string_view is_special_list[] = {"http", " ",   "https", "ws",
-                                                "ftp",  "wss", "file",  " "};
+static constexpr std::string_view is_special_list[] = {
+    "http", " ", "https", "ws", "ftp", "wss", "file", " "};
+
 // for use with get_special_port
-constexpr uint16_t special_ports[] = {80, 0, 443, 80, 21, 443, 0, 0};
+static constexpr uint16_t special_ports[] = {80, 0, 443, 80, 21, 443, 0, 0};
 }  // namespace details
 
 ada_really_inline constexpr bool is_special(std::string_view scheme) {
   if (scheme.empty()) {
     return false;
   }
-  int hash_value = (2 * scheme.size() + (unsigned)(scheme[0])) & 7;
+  auto const hash_value = static_cast<int>(
+      (2 * static_cast<int>(scheme.size()) + static_cast<unsigned>(scheme[0])) &
+      7U);
   const std::string_view target = details::is_special_list[hash_value];
   return (target[0] == scheme[0]) && (target.substr(1) == scheme.substr(1));
 }
 constexpr uint16_t get_special_port(std::string_view scheme) noexcept {
   if (scheme.empty()) {
     return 0;
   }
-  int hash_value = (2 * scheme.size() + (unsigned)(scheme[0])) & 7;
-  const std::string_view target = details::is_special_list[hash_value];
+  auto const hash_value = static_cast<int>(
+      (2 * scheme.size() + static_cast<unsigned>(scheme[0])) & 7U);
+  std::string_view const target = details::is_special_list[hash_value];
```

Comment:
You are replacing 

```
int hash_value = (2 * scheme.size() + (unsigned)(scheme[0])) & 7;
```

with

```
auto const hash_value = static_cast<int>(
       (2 * scheme.size() + static_cast<unsigned>(scheme[0])) & 7U);
```

When would the two expressions differ and why is the second one better/more correct?

Reply:
It's the same thing as the one above, the use of `auto` is there because an additional cast is used to make sure we're avoiding implicit casts and everything explicitly said, and also we don't want to repeat ourselves either, thus the use of `auto`.